source/userguide/entity_resolver.rst

```diff
@@ -171,56 +171,243 @@ Create the synonyms and make them available to your MindMeld app, going entity t
 
-Once all of the entity mapping files are generated, you can either (1) train the resolver as a standalone component, or (2) build the whole NLP pipeline, which trains the resolver along with the other components.
+.. note::
 
-Train the resolver alone
-^^^^^^^^^^^^^^^^^^^^^^^^
+   If you choose not to use Elasticsearch, MindMeld now provides alternate choices for entity resolution. See :ref:`Resolver configurations <resolver_configurations>` below for specifying a custom resolver configuration in your app.
 
-To train the resolver as a standalone component, adapt the following snippet to the particulars of your app:
+One can simply build all the entity resolvers required for an app in one-go by using :meth:`NaturalLanguageProcessor.build()`. Among its other tasks, this initializes as well as fits the required resolvers. Later on, when you run :meth:`NaturalLanguageProcessor.process()`, the NLP pipeline includes the resolved entities in its results.
 
 .. code-block:: python
 
   from mindmeld import configure_logs; configure_logs()
   from mindmeld.components.nlp import NaturalLanguageProcessor
   nlp = NaturalLanguageProcessor(app_path='food_ordering')
-  nlp.domains['ordering'].intents['build_order'].build()
+  nlp.build()
+
+Upon building all the required resolvers, to access a particular resolver, adapt the following snippet to the particulars of your app:
+
+.. code-block:: python
+
   er = nlp.domains['ordering'].intents['build_order'].entities['dish'].entity_resolver
-  er.fit()
+  er
+
+.. code-block:: console
+
+  <ElasticsearchEntityResolver ready: True, dirty: False>
+
+Access & train an entity resolver as standalone component
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Accessing entity resolvers and training them indepenedent of the :mod:`nlp` pipeline is sometimes required, for example, to evaluate performances of different resolvers on your dataset(s). When you are ready to begin experimenting, import `EntityResolverFactory` as follows:
+
+.. code-block:: python
+
+   from mindmeld.components import EntityResolverFactory
+   er = EntityResolverFactory.create_resolver(app_path='food_ordering', entity_type='dish')
+   er
+
+.. code-block:: console
+
+   <ElasticsearchEntityResolver ready: False, dirty: False>
+
+If no resolver configs are specified while creating a resolver, as mentioned previously, the module loads one based on default configs, which involves `Elasticsearch <https://www.elastic.co/products/elasticsearch>`_ based entity resolution. Instead, one can specify the type of resolver they would like to use in the arguments of `create_resolver()` as follows:
```

Comment:
Line 222-237 seems redundant since we explain resolver configs in detail in the next section.